odiglet/odiglet.go

```diff
@@ -189,13 +192,45 @@ func OdigletInitPhase(clientset *kubernetes.Clientset) {
 		os.Exit(-1)
 	}
 
-	odigletInstalledLabel := k8snode.DetermineNodeOdigletInstalledLabelByTier()
+	ctx := context.Background()
 
-	log.Logger.V(0).Info("Adding Label to Node", "odigletLabel", odigletInstalledLabel)
+	// Get the Odigos configuration, retry on odigos effective config not found error
+	var odigosConfig common.OdigosConfiguration
+	err = retry.OnError(retry.DefaultBackoff, func(err error) bool {
+		return errors.Is(err, k8sutils.ErrOdigosEffectiveConfigNotFound)
+	}, func() error {
+		var fetchErr error
+		odigosConfig, fetchErr = k8sutils.GetCurrentOdigosConfigWithClientset(ctx, clientset)
+		return fetchErr
+	})
 
-	if err := k8snode.AddLabelToNode(clientset, nn, odigletInstalledLabel, k8sconsts.OdigletInstalledLabelValue); err != nil {
-		log.Logger.Error(err, "Failed to add Odiglet installed label to the node")
-		os.Exit(-1)
+	if err != nil {
+		if errors.Is(err, k8sutils.ErrOdigosEffectiveConfigNotFound) {
+			log.Logger.Info("Odigos effective config not found after retries, skipping taint removal")
+		} else {
+			log.Logger.Error(err, "Failed to fetch Odigos configuration")
+			os.Exit(-1)
+		}
+	}
+
+	// If Karpenter is enabled, remove the startup taint from the node
+	if odigosConfig.KarpenterEnabled != nil && *odigosConfig.KarpenterEnabled {
+		log.Logger.Info("KarpenterEnabled is true, attempting to remove startup taint")
+		if err := k8snode.RemoveStartupTaint(clientset, nn); err != nil {
+			log.Logger.Error(err, "Failed to remove startup taint from node")
+			os.Exit(-1)
+		} else {
+			log.Logger.Info("Successfully removed startup taint from node")
+		}
+	} else {
+		// Karpenter is not enabled, add the odiglet installed label to the node <Default behavior>
```

Comment:
can we also add this label with Karpetner?
will it do any harm, givin that the webhook will not set any taints for it?